bondcpp/src/bond.cpp

```diff
@@ -437,53 +428,51 @@ bool Bond::waitUntilBroken(rclcpp::Duration timeout)
       break;  //  The deadline has expired
     }
     r.sleep();
+
+    std::unique_lock<std::mutex> lock(state_machine_mutex_);
+    if (sm_.getState().getId() == SM::Dead.getId()) {
+      broken = true;
+      break;
+    }
   }
 
-  return sm_.getState().getId() == SM::Dead.getId();
+  return broken;
 }
 
 bool Bond::isBroken()
 {
-  std::unique_lock<std::mutex> lock(mutex_);
+  std::unique_lock<std::mutex> lock(state_machine_mutex_);
   return sm_.getState().getId() == SM::Dead.getId();
 }
 
 void Bond::breakBond()
 {
-  {
-    std::unique_lock<std::mutex> lock(mutex_);
-    if (sm_.getState().getId() != SM::Dead.getId()) {
-      sm_.Die();
-      publishStatus(false);
-    }
+  std::unique_lock<std::mutex> lock(state_machine_mutex_);
+  if (sm_.getState().getId() != SM::Dead.getId()) {
+    sm_.Die();
+    publishStatus(false);
   }
   flushPendingCallbacks();
```

Comment:
Can we drop the `state_machine_mutex` before calling `flushPendingCallbacks`?  Theoretically we don't need to hold it across that.  Same goes for the rest of the `on..Timeout()` methods below.

Reply:
No, I will add the scopes back in for those.